tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -63,6 +66,699 @@ public function test_wp_enqueue_script() {
+		$expected_header .= "<script type='text/javascript' src='/header-new.js' id='header-new-js'></script>\n";
+		$expected_header .= "<script type='text/javascript' src='/enqueue-header-old.js' id='enqueue-header-old-js'></script>\n";
+		$expected_header .= "<script type='text/javascript' src='/enqueue-header-new.js' id='enqueue-header-new-js'></script>\n";
+
+		$expected_footer  = "<script type='text/javascript' src='/footer-old.js' id='footer-old-js'></script>\n";
+		$expected_footer .= "<script type='text/javascript' src='/footer-new.js' id='footer-new-js'></script>\n";
+		$expected_footer .= "<script type='text/javascript' src='/enqueue-footer-old.js' id='enqueue-footer-old-js'></script>\n";
+		$expected_footer .= "<script type='text/javascript' src='/enqueue-footer-new.js' id='enqueue-footer-new-js'></script>\n";
+
+		$this->assertSame( $expected_header, $header );
+		$this->assertSame( $expected_footer, $footer );
+	}
+
+	/**
+	 * Test normalized script args.
+	 *
+	 * @ticket 12009
+	 */
+	public function test_get_normalized_script_args() {
+		global $wp_scripts;
+		$args = array(
+			'in_footer' => true,
+			'strategy'  => 'async',
+		);
+		wp_enqueue_script( 'footer-async', '/footer-async.js', array(), null, $args );
+		$this->assertSame( $args, $wp_scripts->get_data( 'footer-async', 'script_args' ) );
+
+		// Test defaults.
+		$expected_args = array(
+			'in_footer' => true,
+			'strategy'  => 'blocking',
+		);
+		wp_register_script( 'defaults-strategy', '/defaults.js', array(), null, array( 'in_footer' => true ) );
+		$this->assertSame( $expected_args, $wp_scripts->get_data( 'defaults-strategy', 'script_args' ) );
+
+		$expected_args = array(
+			'in_footer' => false,
+			'strategy'  => 'async',
+		);
+		wp_register_script( 'defaults-in-footer', '/defaults.js', array(), null, array( 'strategy' => 'async' ) );
+		$this->assertSame( $expected_args, $wp_scripts->get_data( 'defaults-in-footer', 'script_args' ) );
+
+		// scripts_args not set of args parameter is empty.
+		wp_register_script( 'empty-args-array', '/defaults.js', array(), null, array() );
+		$this->assertSame( false, $wp_scripts->get_data( 'defaults', 'script_args' ) );
+
+		wp_register_script( 'no-args', '/defaults.js', array(), null );
+		$this->assertSame( false, $wp_scripts->get_data( 'defaults-no-args', 'script_args' ) );
+
+		// Test backward compatibility.
+		$expected_args = array(
+			'in_footer' => true,
+			'strategy'  => 'blocking',
+		);
+		wp_enqueue_script( 'footer-old', '/footer-async.js', array(), null, true );
+		$this->assertSame( $expected_args, $wp_scripts->get_data( 'footer-old', 'script_args' ) );
+	}
+
+	/**
+	 * Test script strategy doing it wrong.
+	 *
+	 * For an invalid strategy defined during script registration, default to a blocking strategy.
+	 *
+	 * @ticket 12009
+	 */
+	public function test_script_strategy_doing_it_wrong() {
+		$this->setExpectedIncorrectUsage( 'WP_Scripts::get_intended_strategy' );
```

Comment:
Use the `@expectedIncorrectUsage WP_Scripts::get_intended_strategy` annotation in the docblock instead. [See here](https://core.trac.wordpress.org/ticket/56070) for the order and spacing of test method documentation annotations.

Reply:
Done in 88d52238207893aa8b55787ed5979dcb990f3e4f